File: C++/Day-4/BankAccount.cpp
#include<iostream>
#include<string.h>
using namespace std;

class BankAccount {
	
	private :
		string name;
		int accountNumber;
		double balance;
		
	public :
		
		BankAccount(){
			cout<<"Default call"<<endl;
		}
		
		BankAccount(string name, int accountNumber, double balance){
			this->name = name;
			this->accountNumber = accountNumber;
			this->balance = balance;
		}
	
	void setName(string name){
		this->name = name;
	}
	
	void setAccountNumber(int accountNumber){
		this->accountNumber = accountNumber;
	}
	
	void setBalance(double balance){
		this->balance = balance;
	}
	
	string getName(){
		return name;
	}
	
	int getAccountNumber(){
		return this->accountNumber;
	}
	
	double getBalance(){
		return balance;
	}
	
	
	void sendMoney(double money, BankAccount &toAcc){
		toAcc.balance = toAcc.balance + money;
		balance = balance - money;
	}
	
	void depositeMoney(double money){
		balance = balance + money;
	}
	
	void displayBalance(){
		cout<<"Account balance = "<<balance<<" for Account holder Name = "<<name<<endl;
	}
};


int main(){
	
	BankAccount ba1;
	ba1.setName("Pranay");
	ba1.setAccountNumber(123);
	ba1.setBalance(10000);
	
	ba1.displayBalance();
	
	BankAccount ba2;
	ba2.setName("Anuj");
	ba2.setAccountNumber(456);
	ba2.setBalance(15000);
	ba2.displayBalance();
	ba2.sendMoney(4000,ba1);
	
	ba1.displayBalance();
	ba2.displayBalance();
	
	ba1.depositeMoney(20000);
	ba1.displayBalance();
	
}

File: JAVA/Revision-AT/Day-1/Arrays_Problem/src/com/Question_3/BankAccount.java
package com.Question_3;

//        a. Define a class named BankAccount with the following attributes:
//        i. accountNumber: Unique identifier for each bank account.
//        ii. accountHolderName: Name of the account holder.
//        iii. balance: Current balance in the account.
//        b. Implement appropriate constructors and methods for the BankAccount class:
//        i. Include methods to deposit money into the account, withdraw money from
//        the account, and check the account balance.
//        ii. Ensure that the balance cannot be negative after a withdrawal.
//        c. Create a main program to interact with the BankAccount class:
//        i. Allow users to perform operations such as depositing money, withdrawing
//        money, and checking their account balance using a simple menu-driven interface.
//        ii. Prompt users to enter the necessary information for each operation (e.g.,
//        account number, account holder name, amount to deposit/withdraw).
//        iii. Display appropriate messages to indicate the success or failure of each
//        operation.
public class BankAccount {

    private int accountNumber;
    private String accountHolderName;
    private double balance;

    public BankAccount() {
        accountNumber = 0;
        accountHolderName = "NIL";
        balance = 0.0;
    }

    public BankAccount(int accountNumber, String accountHolderName, double balance) {
        this.accountNumber = accountNumber;
        this.accountHolderName = accountHolderName;
        this.balance = balance;
    }

    public int getAccountNumber() {
        return accountNumber;
    }

    public void setAccountNumber(int accountNumber) {
        this.accountNumber = accountNumber;
    }

    public String getAccountHolderName() {
        return accountHolderName;
    }

    public void setAccountHolderName(String accountHolderName) {
        this.accountHolderName = accountHolderName;
    }

    public double getBalance() {
        return balance;
    }

    public void setBalance(double balance) {
        this.balance = balance;
    }

    @Override
    public String toString() {
        return "Account Number = " + accountNumber +
                "\nAccount Holder Name = '" + accountHolderName + '\'' +
                "\nBalance = " + balance;
    }
}
